test-cargo-miri/src/lib.rs

```diff
@@ -3,5 +3,11 @@
 /// assert!(cargo_miri_test::make_true());
 /// ```
 pub fn make_true() -> bool {
+    extern "C" {
+        fn use_cdylib();
+    }
+    drop(use_cdylib);
```

Comment:
Hm, I don't think this will actually "use" anything in Miri.

Reply:
Would coercing it to a function pointer be better? (I think calling it is not supported since it's `extern "C"`.)

Reply:
That won't do much either I think -- a call would be needed.
Since that doesn't work... I am not sure if we can't really properly test this currently.

Reply:
Maybe it's better to remove this, the function can't be called anyway. The `cdylib` dependency itself can still be used to test whether `cargo-miri` can build it or not.

Reply:
So, before this PR that cdylib library caused some kind of error?

Reply:
No, but `cargo-miri` used to handle `cdylib` specially so I wrote this test.

(I explained that in https://github.com/rust-lang/miri/pull/1710#discussion_r575829920, looks like it got lost:)
> I also added a test for `cdylib`-only crate (since that is the "skipped" case if this is not removed, but this is removed anyway, so that test may be redundant).

Reply:
Okay, sure.
Please add a comment somewhere in that crate to explain why it exists.

Reply:
Should I remove this probably useless `drop(use_cdylib)` then?

Reply:
Yeah, please do.